gslib/tests/test_acl.py

```diff
@@ -760,3 +769,181 @@ class TestAclOldAlias(TestAcl):
+                       ' --predefined-acl=private gs://bucket').format(
+                           os.path.join('fake_dir', 'bin', 'gcloud')),
+                      info_lines)
+
+  @mock.patch.object(acl.AclCommand, 'RunCommand', new=mock.Mock())
+  def test_shim_translates_acl_set_multiple_buckets_urls(self):
+    with SetBotoConfigForTest([('GSUtil', 'use_gcloud_storage', 'True'),
+                               ('GSUtil', 'hidden_shim_mode', 'dry_run')]):
+      with SetEnvironmentForTest({
+          'CLOUDSDK_CORE_PASS_CREDENTIALS_TO_GSUTIL': 'True',
+          'CLOUDSDK_ROOT_DIR': 'fake_dir',
+      }):
+        mock_log_handler = self.RunCommand('acl', [
+            'set', '-f', 'acl-file', 'gs://bucket', 'gs://bucket1',
+            'gs://bucket2'
+        ],
+                                           return_log_handler=True)
+        info_lines = '\n'.join(mock_log_handler.messages['info'])
+        self.assertIn(('Gcloud Storage Command: {} alpha storage buckets update'
+                       ' --acl-file=acl-file --continue-on-error'
+                       ' gs://bucket gs://bucket1 gs://bucket2').format(
+                           os.path.join('fake_dir', 'bin', 'gcloud')),
+                      info_lines)
+
+  @mock.patch.object(acl.AclCommand, 'RunCommand', new=mock.Mock())
+  def test_shim_translates_acl_set_multiple_objects_urls(self):
+    with SetBotoConfigForTest([('GSUtil', 'use_gcloud_storage', 'True'),
+                               ('GSUtil', 'hidden_shim_mode', 'dry_run')]):
+      with SetEnvironmentForTest({
+          'CLOUDSDK_CORE_PASS_CREDENTIALS_TO_GSUTIL': 'True',
+          'CLOUDSDK_ROOT_DIR': 'fake_dir',
+      }):
+        mock_log_handler = self.RunCommand('acl', [
+            'set', '-f', 'acl-file', 'gs://bucket/object',
+            'gs://bucket/object1', 'gs://bucket/object2'
+        ],
+                                           return_log_handler=True)
+        info_lines = '\n'.join(mock_log_handler.messages['info'])
+        self.assertIn(
+            ('Gcloud Storage Command: {} alpha storage objects update'
+             ' --acl-file=acl-file --continue-on-error gs://bucket/object'
+             ' gs://bucket/object1 gs://bucket/object2').format(
+                 os.path.join('fake_dir', 'bin', 'gcloud')), info_lines)
+
+  @mock.patch.object(acl.AclCommand, 'RunCommand', new=mock.Mock())
+  def test_shim_translates_acl_set_multiple_buckets_urls_recursive_all_versions(
+      self):
+    with SetBotoConfigForTest([('GSUtil', 'use_gcloud_storage', 'True'),
+                               ('GSUtil', 'hidden_shim_mode', 'dry_run')]):
+      with SetEnvironmentForTest({
+          'CLOUDSDK_CORE_PASS_CREDENTIALS_TO_GSUTIL': 'True',
+          'CLOUDSDK_ROOT_DIR': 'fake_dir',
+      }):
+        mock_log_handler = self.RunCommand('acl', [
+            'set', '-r', '-a', 'acl-file', 'gs://bucket', 'gs://bucket1',
```

Comment:
Change one of these to an object URL to test that mix of buckets and objects is allowed if recursion is enabled.